src/Induction/WellFounded.agda

```diff
@@ -79,6 +89,26 @@ module All {a lt} {A : Set a} {_<_ : Rel A lt}
   Please use wfRecBuilder instead."
   #-}
 
+module FixPoint
+  {a ℓ} {A : Set a} {_<_ : Rel A ℓ} (wf : WellFounded _<_)
+  (P : Pred A ℓ) (f : WfRec _<_ P ⊆′ P)
+  (f-ext : (x : A) {IH IH' : WfRec _<_ P x} → (∀ {y} y<x → IH y y<x ≡ IH' y y<x) → f x IH ≡ f x IH')
+  where
+
+  someWfRecInvar : ∀ x → (q q' : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q'
+  someWfRecInvar = All.wfRec wf (a ⊔ ℓ)
+                             (λ x → (q q' : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q')
+                             (λ { x IH (acc rs) (acc rs') → f-ext x (λ y<x → IH _ y<x (rs _ y<x) (rs' _ y<x)) })
+
```

Comment:
A similar lemma is called `NAME-irrelevant` in [`Data.Fin.Propertiess`](https://agda.github.io/agda-stdlib/Data.Fin.Properties.html#12753)